files/usr/sbin/jeos-config

```diff
@@ -0,0 +1,147 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+. /etc/os-release
+. "/usr/share/jeos-firstboot/jeos-firstboot-functions"
+. "/usr/share/jeos-firstboot/jeos-firstboot-dialogs"
+
+# for testing we may run as non root
+if [ -w /run ]; then
+	export TMPDIR=/run
+	# debugging
+	if [ -n "$FIRSTBOOT_DEBUG" ]; then
+		set -x
+		exec 2>/var/log/firstboot-debug
+	fi
+else
+	dry=1
+fi
+
+if [ -n "$dry" ]; then
+	run() {
+		echo "$@"
+	}
+else
+	run() {
+		"$@"
+	}
+fi
+
+if pushd "/usr/share/jeos-firstboot/modules" &>/dev/null; then
+	for module in *; do
+		if [ -f "${module}" ] && source "${module}"; then
+			modules+=("${module}")
+		fi
+	done
+	popd &>/dev/null
+fi
+
+cleanup() {
+	#call_module_hook cleanup
+	echo .oOo.oOo.oOo. > $dialog_out
+	rm -f "$dialog_out"
+	# reenable systemd and kernel logs
+	run kill -s SIGRTMAX-10 1
+	run setterm -msg on 2>/dev/null || true
+}
+trap cleanup EXIT
+
+# If the configuration is not loaded and we are in the first terminal
+# instance, make sure that the variables are declared.
+JEOS_LOCALE=${JEOS_LOCALE-}
+JEOS_KEYTABLE=${JEOS_KEYTABLE-}
+
+select_config()
+{
+	modules_order=("locale" '' "keytable" '' "timezone" '' "password" '' "network" '')
+	for module in "${modules[@]}"; do
+		modules_order+=("${module}" '')
+	done
+
+	d --menu $"Select config module" 0 0 $dh_menu "${modules_order[@]}"
+}
+
+usage()
+{
+cat <<EOF
+Usage: jeos-config [OPTION...] [CONFIG_NAME]
+Configure system settings using an interactive dialog
+
+	-h		shows this usage help
+	locale		Show configuration for locale
+	keytable	Show configuration for keyboard
+	timezone	Show configuration for timezone
+	password	Show configuration for password
+	network		Show configuration for network
+$(for module in "${modules[@]}"; do
+	echo "	${module}	Show configuration for ${module}"
+done)
+EOF
+}
+
+while getopts ":h" opt; do
+	case ${opt} in
+		h) 
+			usage
+			exit 0
+			;;
+		\?) 
+			echo "Invalid Option: -$OPTARG" 1>&2
+			usage
+     			exit 1
+			;;
+	esac
+done
+
+if [ $# -eq 0 ]; then
+	select_config
+	subcommand=${result}
+else
+	subcommand=$1; shift
+fi
+
+case "$subcommand" in
+
+	locale)
+		dialog_locale
+		apply_locale
+        	;;
+	keytable)
+		dialog_keytable
+		get_current_locale
+		apply_locale_and_keytable
+		;;	
+	timezone)
+		dialog_timezone
+		timedatectl set-timezone "$JEOS_TIMEZONE"
+		;;	
+	network)
+		dialog_network
+		d --infobox $"Restarting network ..." 3 26 || true
+		systemctl restart network
+		;;
+	password)
+		dialog_password
+		apply_password
+		;;
+	*)
+		call_module_config "$subcommand"|| echo "Unknown option"
```

Comment:
We need a generic `call_module` which is just `call_module_hook` for a single module, then this can be just

`call_module "$subcommand" jeos_config`